server/etcdserver/version/version.go

```diff
@@ -65,3 +77,80 @@ func detectStorageVersion(lg *zap.Logger, tx backend.ReadTx) (*semver.Version, e
 	copied := V3_5
 	return &copied, nil
 }
+
+func migrateOnce(lg *zap.Logger, tx backend.BatchTx, current semver.Version, upgrade bool) (*semver.Version, error) {
+	var target semver.Version
+	if upgrade {
+		target = semver.Version{Major: current.Major, Minor: current.Minor + 1}
+	} else {
+		target = semver.Version{Major: current.Major, Minor: current.Minor - 1}
+	}
+	ms := migrations(lg, current, target)
+	err := runMigrations(tx, target, ms, upgrade)
+	if err != nil {
+		return nil, err
+	}
+	return &target, nil
+}
+
+func migrations(lg *zap.Logger, current semver.Version, target semver.Version) []Migration {
+	var schemaKey semver.Version
+	// Migrations should be taken from higher version
+	if current.LessThan(target) {
+		schemaKey = target
+	} else {
+		schemaKey = current
+	}
+	ms, found := schema[schemaKey]
+	if !found {
+		lg.Panic("version is not supported", zap.String("storage-version", target.String()))
+	}
+	return ms
+}
+
+var schema = map[semver.Version][]Migration{
+	V3_6: {
+		&newField{bucket: buckets.Meta, fieldName: buckets.MetaStorageVersionName, fieldValue: []byte("")},
+	},
+}
+
+type Migration interface {
+	UnsafeUpgrade(backend.BatchTx) error
+	UnsafeDowngrade(backend.BatchTx) error
+}
+
+type newField struct {
```

Comment:
Bit unsure what this struct name conveys here but can't think of a better suggestion. 🤔  Maybe just `field`?

Reply:
This struct doesn't represent a field, but an action of adding a field. Would name `addNewField` be better?

Reply:
Sounds good to me 👍